File: apps/android/feature/insights/domain/src/main/java/com/scribblefit/feature/insights/domain/repository/InsightsRepository.kt
package com.scribblefit.feature.insights.domain.repository

import com.scribblefit.core.model.AIInsight
import com.scribblefit.feature.insights.domain.model.FrequencyData
import com.scribblefit.feature.insights.domain.model.MuscleGroupDistribution
import com.scribblefit.feature.insights.domain.model.VolumeDataPoint
import kotlinx.coroutines.flow.Flow

interface InsightsRepository {
    fun getVolumeInsights(startDate: Long, endDate: Long): Flow<List<VolumeDataPoint>>
    fun getFrequencyInsights(startDate: Long, endDate: Long): Flow<FrequencyData>
    fun getMuscleDistributionInsights(
        startDate: Long,
        endDate: Long
    ): Flow<List<MuscleGroupDistribution>>

    fun getAIOverview(startDate: Long, endDate: Long): Flow<List<AIInsight>>
}

File: apps/ios/ScribbleFit/Sources/Domain/Repositories/InsightsRepository.swift
import Foundation

@MainActor
public protocol InsightsRepository: Sendable {
    func getVolumeInsights(startDate: Date, endDate: Date) -> AsyncStream<[VolumeDataPoint]>
    func getFrequencyInsights() -> AsyncStream<FrequencyData>
    func getMuscleDistributionInsights() -> AsyncStream<[MuscleGroupDistribution]>
    func getAIOverview() async throws -> AIOverview
}
